### PR TITLE
[Core] Less restrictive check of orthonormal basis

### PR DESCRIPTION
**Pull request by loumalouomega**
@jcotela  had problems with this

### CODE REVIEW

**Review by jcotela — DISMISSED** [review]
Ok for me, but if you want a vector of norm ~ 1, I would put "unit vector" in the error message. By the way, at least with amatrix (never checked ublas), there should be a normalize() method to automatically correct your input.

**Comment by loumalouomega** [comment]
OK

**Comment by loumalouomega** [comment]
I found an issue

File: kratos/utilities/math_utils.h [hunk 1]
```diff
@@ -976,34 +978,43 @@ class MathUtils
...
-        const double norm_a = norm_2(a);
-        KRATOS_DEBUG_ERROR_IF_NOT(norm_a < std::numeric_limits<double>::epsilon()) << "Zero norm of the vector" << std::endl;
-        a /= norm_a;
-        b[0] = - (c[0] * c[1])/(1.0 + c[2]);
-        b[1] = 1.0 - std::pow(c[1], 2)/(1.0 + c[2]);
-        b[2] = -c[1];
-        const double norm_b = norm_2(b);
-        KRATOS_DEBUG_ERROR_IF_NOT(norm_b < std::numeric_limits<double>::epsilon()) << "Zero norm of the vector" << std::endl;
-        b /= norm_b;
+        if ((c[2] + 1.0) > ZeroTolerance) {
```
Comment (jcotela) [inline kratos/utilities/math_utils.h]: std::abs? Is it a problem if c[2] = -0.99999.....?
also, note that comparing abs(a-b) > epsilon is essentially equivalent to a == b, so maybe we need to be a bit more relaxed with the tolerance (up to you, though, I didn't study the complete algorithm)

Comment (loumalouomega) [reply kratos/utilities/math_utils.h]: This covers all the cases (this must be activated only for [0,0,-1])

**Comment by loumalouomega** [comment]
@jcotela do you need anything else here?

**Comment by RiccardoRossi** [comment]
@loumalouomega the hughes_moeller version looks much more robust to me than the modified one wrt to precision... i have the feeling we should use it by default even if it is more costly 

i mean... i would definitely give a premium to robustness over speed...

**Comment by loumalouomega** [comment]
What do you mean @RiccardoRossi, I just added an exception case.

**Review by frastellini — DISMISSED** [review]
Ok for me

**Comment by loumalouomega** [comment]
@RiccardoRossi this is done

**Review by ipouplana — APPROVED** [review]
Good

**Pull request merged**